### ISSUE TITLE
NameError: name 'Cookie' is not defined

### ISSUE THREAD

**Issue by yesh-r** [UCL-INGI/INGInious#208]
Hi,

I've followed all the instructions and installed the app. When I create the ```inginious-webapp``` everything works fine until I open my browser at localhost:8080. Once I load the browser throws a 500 response error and the terminal shows the error mentioned above. I'm also attaching the log.

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/web/wsgiserver/wsgiserver3.py", line 1089, in communicate
    req.respond()
  File "/usr/local/lib/python3.5/dist-packages/web/wsgiserver/wsgiserver3.py", line 877, in respond
    self.server.gateway(self).respond()
  File "/usr/local/lib/python3.5/dist-packages/web/wsgiserver/wsgiserver3.py", line 1980, in respond
    response = self.req.server.wsgi_app(self.env, self.start_response)
  File "/usr/local/lib/python3.5/dist-packages/inginious/common/log.py", line 47, in __call__
    return self.app(environ, xstart_response)
  File "/usr/local/lib/python3.5/dist-packages/inginious/frontend/common/static_middleware.py", line 64, in __call__
    return self.app(environ, start_response)
  File "/usr/local/lib/python3.5/dist-packages/web/application.py", line 295, in wsgi
    result = self.handle_with_processors()
  File "/usr/local/lib/python3.5/dist-packages/web/application.py", line 267, in handle_with_processors
    return process(self.processors)
  File "/usr/local/lib/python3.5/dist-packages/web/application.py", line 264, in process
    raise self.internalerror()
  File "/usr/local/lib/python3.5/dist-packages/web/application.py", line 545, in internalerror
    return debugerror()
  File "/usr/local/lib/python3.5/dist-packages/web/debugerror.py", line 313, in debugerror
    return web._InternalError(djangoerror())
  File "/usr/local/lib/python3.5/dist-packages/web/debugerror.py", line 303, in djangoerror
    return t(exception_type, exception_value, frames)
  File "/usr/local/lib/python3.5/dist-packages/web/template.py", line 904, in __call__
    return BaseTemplate.__call__(self, *a, **kw)
  File "/usr/local/lib/python3.5/dist-packages/web/template.py", line 831, in __call__
    return self.t(*a, **kw)
  File "/usr/local/lib/python3.5/dist-packages/web/debugerror.py", line 196, in __template__
    </ul>
  File "/usr/local/lib/python3.5/dist-packages/web/webapi.py", line 487, in cookies
    ctx._parsed_cookies = parse_cookies(http_cookie)
  File "/usr/local/lib/python3.5/dist-packages/web/webapi.py", line 447, in parse_cookies
    cookie = Cookie.SimpleCookie()
NameError: name 'Cookie' is not defined
```

**Comment by anthonygego** [comment]
It seems the issue comes from web.py. What is your Linux distribution and the version used ? Did you install INGInious using the Github master branch ? What is the web.py version installed by pip ? Did you simply try reinstalling the packages ?

**Comment by GuillaumeDerval** [comment]
https://github.com/webpy/webpy/blob/master/web/webapi.py#L447
https://docs.python.org/2/library/cookie.html

Cookie has been renamed to http.cookie in Py3, but this has not been reflected in web.py.

We'll report the problem upstream.

The real question is: what is your browser? We have been running on web.py on Py3 for a long time now and nobody ever used this particular part of code (or reported the bug). Your browser is doing something unusual :D

**Comment by sizhky** [comment]
I was using chromium browser in ubuntu 16.04. Yeah, I have installed the latest master branch. I did reinstall all the packages.
@GuillaumeDerval Thanks, I will try with a different browser and get back to you

**Comment by GuillaumeDerval** [comment]
Reported upstream: https://github.com/webpy/webpy/pull/421

**Comment by GuillaumeDerval** [comment]
Until the fix is merged by web.py maintainers and pushed to Pypi, you can run this command to fix web.py locally:

`pip3 install --upgrade git+https://github.com/GuillaumeDerval/webpy.git@patch-1`

**Comment by yesh-r** [comment]
Now it says ```AttributeError: 'SimpleCookie' object has no attribute 'iteritems'```

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/web/wsgiserver/wsgiserver3.py", line 1089, in communicate
    req.respond()
  File "/usr/local/lib/python3.5/dist-packages/web/wsgiserver/wsgiserver3.py", line 877, in respond
    self.server.gateway(self).respond()
  File "/usr/local/lib/python3.5/dist-packages/web/wsgiserver/wsgiserver3.py", line 1980, in respond
    response = self.req.server.wsgi_app(self.env, self.start_response)
  File "/usr/local/lib/python3.5/dist-packages/inginious/common/log.py", line 47, in __call__
    return self.app(environ, xstart_response)
  File "/usr/local/lib/python3.5/dist-packages/inginious/frontend/common/static_middleware.py", line 64, in __call__
    return self.app(environ, start_response)
  File "/usr/local/lib/python3.5/dist-packages/web/application.py", line 295, in wsgi
    result = self.handle_with_processors()
  File "/usr/local/lib/python3.5/dist-packages/web/application.py", line 267, in handle_with_processors
    return process(self.processors)
  File "/usr/local/lib/python3.5/dist-packages/web/application.py", line 264, in process
    raise self.internalerror()
  File "/usr/local/lib/python3.5/dist-packages/web/application.py", line 545, in internalerror
    return debugerror()
  File "/usr/local/lib/python3.5/dist-packages/web/debugerror.py", line 313, in debugerror
    return web._InternalError(djangoerror())
  File "/usr/local/lib/python3.5/dist-packages/web/debugerror.py", line 303, in djangoerror
    return t(exception_type, exception_value, frames)
  File "/usr/local/lib/python3.5/dist-packages/web/template.py", line 904, in __call__
    return BaseTemplate.__call__(self, *a, **kw)
  File "/usr/local/lib/python3.5/dist-packages/web/template.py", line 831, in __call__
    return self.t(*a, **kw)
  File "/usr/local/lib/python3.5/dist-packages/web/debugerror.py", line 196, in __template__
    </ul>
  File "/usr/local/lib/python3.5/dist-packages/web/webapi.py", line 487, in cookies
    ctx._parsed_cookies = parse_cookies(http_cookie)
  File "/usr/local/lib/python3.5/dist-packages/web/webapi.py", line 459, in parse_cookies
    cookies = dict([(k, unquote(v.value)) for k, v in cookie.iteritems()])
AttributeError: 'SimpleCookie' object has no attribute 'iteritems'
```

**Comment by yesh-r** [comment]
@GuillaumeDerval You are right. It is working on Firefox. I still wonder what makes chromium throw a 500 error though...

**Comment by GuillaumeDerval** [comment]
I removed the lost iteritems in my patch; if you still want to use chromium, it should work now if you reuse my branch.

**Comment by yesh-r** [comment]
Thanks for the help @GuillaumeDerval

**Comment by GuillaumeDerval** [comment]
Let's not close this until the patch is merged with web.py

**Comment by GuillaumeDerval** [comment]
Web.py is going to kill me/us, and it's time to clean the issues. I'm closing it, Google should be good enough to find this issue if somebody else needs the fix.

**Comment by mbifulco** [comment]
👀 Near as I can tell, the fix for this is an [unmerged PR](https://github.com/webpy/webpy/pull/419) on webpy?  Does that fix the compatibility issue here and in #257?

**Comment by GuillaumeDerval** [comment]
Yes. We have to wait for the maintainers of web.py.

I just had the idea to try to monkey-patch web.py to fix the bug and it seems that it works. Fix is at #309. I'll need to discuss with @anthonygego to see if we merge the PR or not, but for now you have an INGInious-only-fix.
